Stix adapter guide

jlstrick83 opened the pull request.

delliott90 commented on adapter-guide/develop-stix-adapter.md:
```diff
@@ -0,0 +1,602 @@
+
+* [Introduction](stix-adapter-overview.md)
+* [Scenario](#scenario)
+* [Prerequisites](#prerequisites)
```
The Prerequisites link doesn't work. I think it's because this actually links to the heading (denoted with the \#\#) which is Prerequisites for Developer A. The tags, such as {: \#scenario} don't seem to do anything and they are actually visible when viewing it as the markdown.

delliott90 commented on adapter-guide/develop-stix-adapter.md:
```diff
@@ -0,0 +1,602 @@
+By implementing a new adapter, *Developer A* allows *Product A* to fit into the workflow.  
+
+
+## Prerequisites for *Developer A*
```
As per the comment above, I think this just needs to be \#\# Prerequisites for the above link to work.

delliott90 commented on adapter-guide/develop-stix-adapter.md:
```diff
@@ -0,0 +1,602 @@
+### Create a Translation module
+{: #translation-mod}
+
+1. [Create a translation module folder](#translation-1)
```
None of these links seem to work.

jlstrick83 replied:
Noted Danny. I'll make the corrections for all of the links.

delliott90 commented on adapter-guide/develop-stix-adapter.md:
````diff
@@ -0,0 +1,602 @@
+
+The following STIX pattern:
+
+```"[network-traffic:src_port = 37020 AND network-traffic:dst_port = 635] START '2016-06-
````
This is getting cut off, you need to have the \'\'\' on it's own line.

delliott90 commented on adapter-guide/develop-stix-adapter.md:
```diff
@@ -0,0 +1,602 @@
+
+This image illustrates where the query string is constructed for the data source query.
+
+![](../images/parse-expression-method.png)
```
This doesn't show up in the readme.

delliott90 commented on adapter-guide/develop-stix-adapter.md:
```diff
@@ -0,0 +1,602 @@
+2.	Visually verify that all expected data is in the returned STIX bundle. If a data source field in your sample results is mapped in `to_stix_map.json`, the value must be in the STIX bundle under the mapped STIX property.
+
+**Note:** 
+* The <STIX identity object> represents a data source and is the first observed-data object that gets added to the STIX bundle during results translation. 
```
I think you need to escape the arrow brackets with a back slash. Right now the "STIX identity object" isn't showing up. So it would be \\\< \\\> in the markdown code

delliott90 commented on adapter-guide/develop-stix-adapter.md:
```diff
@@ -0,0 +1,602 @@
+
+**Note:** 
+* The <STIX identity object> represents a data source and is the first observed-data object that gets added to the STIX bundle during results translation. 
+* Each observed-data object, which gets added to the bundle, references the <STIX identity object> to indicate which data source the result came from. 
```
Same as above, arrow brackets need to be escaped.

delliott90 commented on adapter-guide/develop-stix-adapter.md:
````diff
@@ -0,0 +1,602 @@
+    1. Use the following CLI command:    
+        ```
+        python main.py transmit abc '{"host":"<IP address or URL to data source>", "port":"<port number>", "cert": 
+        "-----BEGIN PRIVATE KEY-----<private key>-----END PRIVATE KEY-----\n-----BEGIN CERTIFICATE-----<certificate>-----END CERTIFICATE-----\n"}' '{"auth{"auth": <authentication object>}' status "<Query UUID from test C>"
````
Should now refer to test 4 rather than test C given the change in number of the tests.

delliott90 commented on adapter-guide/develop-stix-adapter.md:
````diff
@@ -0,0 +1,602 @@
+    
+        ```
+        python main.py transmit abc '{"host":"<IP address or URL to data source>", "port":"<port number>", "cert": 
+        "-----BEGIN PRIVATE KEY-----<private key>-----END PRIVATE KEY-----\n-----BEGIN CERTIFICATE-----<certificate>-----END CERTIFICATE-----\n"}' '{"auth{"auth": <authentication object>}' results "<Query UUID from test C>" <Offset Integer> <Length Integer>
````
Change test C to test 4.

delliott90 commented on adapter-guide/develop-stix-adapter.md:
````diff
@@ -0,0 +1,602 @@
+    
+        ```
+        python main.py transmit abc '{"host":"<IP address or URL to data source>", "port":"<port number>", "cert": 
+        "-----BEGIN PRIVATE KEY-----<private key>-----END PRIVATE KEY-----\n-----BEGIN CERTIFICATE-----<certificate>-----END CERTIFICATE-----\n"}' '{"auth": <authentication object>}' delete "<Query UUID from test C>"
````
Change test C to test 4.

delliott90 commented:
I'm thinking the main README.md file should have links at the top to both of your new files. The link from the readme on creating new modules could be repurposed to link to your documents. The current creating new modules section of the readme could probably be removed.

jlstrick83 commented:
Updated the file. Addressed all of the comments mentioned in this request

https://github.com/IBM/stix-shifter/commit/ef0e0716eb26031dec024c4389389edd202cd87a

jlstrick83 commented on README.md:
```diff
@@ -246,33 +247,12 @@ python main.py transmit qradar '{"host":"<ip address>", "port":"<port>", "cert":
-- In `stix-shifter.py`, add `<module-name>` to the `MODULES` array at the top of the file
-- In your module directory, create a new python code file named `<module-name>_translator.py`. This is where you'll be defining your concrete translator class
-  - In `<module-name>_translator.py`, define a class named `Translator` and have it extend `BaseTranslator` from `stix-shifter/src/modules/base/base_translator.py`. (You can use `stix-shifter/src/modules/dummy/dummy_translator.py` as an example)
-  - In `__init__` you need to assign `self.result_translator` and `self.query_translator` to the appropriate query and result translator you want your module to use. For example the QRadar translator uses `JSONToStix` as its result translator and `StixToAQL` as its query translator
```
Do we need to mention this in the guide?

jlstrick83 commented on README.md:
```diff
@@ -246,33 +247,12 @@ python main.py transmit qradar '{"host":"<ip address>", "port":"<port>", "cert":
-  - In `<module-name>_translator.py`, define a class named `Translator` and have it extend `BaseTranslator` from `stix-shifter/src/modules/base/base_translator.py`. (You can use `stix-shifter/src/modules/dummy/dummy_translator.py` as an example)
-  - In `__init__` you need to assign `self.result_translator` and `self.query_translator` to the appropriate query and result translator you want your module to use. For example the QRadar translator uses `JSONToStix` as its result translator and `StixToAQL` as its query translator
-  - You can write your own query and result translators as well, they must be based off of `BaseQueryTranslator` and `BaseResultTranslator` found in `stix-shifter/src/modules/base/`. Again, you can use the dummy module as a decent example on how to setup the concrete classes found in `stix-shifter/src/modules/dummy/`
-- Once you have this all set up you can invoke your module by running `stix_shifter.py` and passing in your translator module name as the first parameter. The second parameter `query or result` determines if your module runs the query or result translator. The third parameter `data` is passed into your translator as the data that will be translated. If you've imported `stix_shifter.py` into other ...
```
This is not mentioned in the (Developing a new STIX-shifter adapter) guide. Should we add it there?

"Once you have this all set up you can invoke your module by running `stix_shifter.py` and passing in your translator module name as the first parameter. The second parameter `query or result` determines if your module runs the query or result translator. The third parameter `data` is passed into your translator as the data that will be translated. If you've imported `stix_shifter.py` into other python code, you can invoke it by running the `translate(module, translation_type, data)` function"
